apis/github

```diff
@@ -17,6 +16,41 @@ local function getAPI(path)
 	end
 end
 
+-- A class for authorization
+local authFile = '.github-auth'
+local Auth = {}
+Auth.__index = Auth
+Auth.new = function(type, user, token)
+	return setmetatable({type=type, user=user, token=token}, Auth)
+end
+Auth.getAll = function()
+	local authTable = {}
+	if fs.exists(authFile) then
+		f = fs.open(authFile, 'r')
+		authTable = textutils.unserialize(f.readAll())
+		f.close()
+	end
+	return authTable
+end
+Auth.get = function(user)
+ local authTable = Auth.getAll()
+ return authTable[user]
+end
+Auth.save = function(self)
+	local authTable = self.getAll()
+	authTable[self.user] = self
+	f = fs.open(authFile, 'w')
+	f.write(textutils.serialize(authTable))
+	f.close()
+end
+Auth.delete = function(user)
```

Comment:
This function assumes a user was deleted even if the user doesn't exist... this shouldn't be an issue but I'm not sure how you feel about that.